enginetest/queries/procedure_queries.go

```diff
@@ -267,6 +267,98 @@ END;`,
 			},
 		},
 	},
+	{
+		Name: "CASE statements",
+		SetUpScript: []string{
+			`CREATE PROCEDURE p1(IN a BIGINT)
+BEGIN
+	DECLARE b VARCHAR(200) DEFAULT "";
+	tloop: LOOP
+		CASE
+			WHEN a < 4 THEN
+				SET b = CONCAT(b, "a");
+				SET a = a + 1;
+			WHEN a < 8 THEN
+				SET b = CONCAT(b, "b");
+				SET a = a + 1;
+			ELSE
+				LEAVE tloop;
```

Comment:
Probably worth testing a case block without an else block and testing the `Case not found` error, too.